internal/services/mobilenetwork/mobile_network_attached_data_network_resource.go

```diff
@@ -0,0 +1,548 @@
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"user_plane_access_ipv4_address": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsIPv4Address,
+		},
+
+		"user_plane_access_ipv4_subnet": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validate.CIDR,
+		},
+
+		"user_plane_access_ipv4_gateway": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsIPv4Address,
+		},
+
+		"user_equipment_address_pool_prefixes": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validate.CIDR,
+			},
+			AtLeastOneOf: []string{
+				"user_equipment_address_pool_prefixes",
+				"user_equipment_static_address_pool_prefixes",
+			},
+		},
+
+		"user_equipment_static_address_pool_prefixes": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validate.CIDR,
+			},
+			AtLeastOneOf: []string{
+				"user_equipment_address_pool_prefixes",
+				"user_equipment_static_address_pool_prefixes",
+			},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r AttachedDataNetworkResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r AttachedDataNetworkResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model AttachedDataNetworkModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MobileNetwork.AttachedDataNetworkClient
+			packetCoreDataPlaneId, err := packetcoredataplane.ParsePacketCoreDataPlaneID(model.MobileNetworkPacketCoreDataPlaneId)
+			if err != nil {
+				return err
+			}
+
+			id := attacheddatanetwork.NewAttachedDataNetworkID(packetCoreDataPlaneId.SubscriptionId, packetCoreDataPlaneId.ResourceGroupName, packetCoreDataPlaneId.PacketCoreControlPlaneName, packetCoreDataPlaneId.PacketCoreDataPlaneName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			attachedDataNetwork := &attacheddatanetwork.AttachedDataNetwork{
+				Location: location.Normalize(model.Location),
+				Properties: attacheddatanetwork.AttachedDataNetworkPropertiesFormat{
+					DnsAddresses:           model.DnsAddresses,
+					UserPlaneDataInterface: attacheddatanetwork.InterfaceProperties{},
+					NaptConfiguration:      expandNaptConfigurationModel(model.NaptConfiguration),
+				},
+				Tags: &model.Tags,
```

Comment:
Shouldn't we use `tags.Expand()` instead?

Reply:
It's already a `map[string]string` object, do we still need to use that?

Reply:
Normally, we define the model using the same type as is used by the underlying plugin SDK, which means the `tags` is of type `map[string]interface{}`